apps/backend/controllers/userController.ts

```diff
@@ -0,0 +1,255 @@
+import { Request, Response } from 'express';
+import supabase from '../config/supabase.js';
+import { UpdateUserBody } from '../types.js';
+
+export async function getUserByID(req: Request, res: Response): Promise<void> {
+    try {
+        const userID = req.params.userid;
+
+        if (!userID) {
+            res.status(400).json({ error: "Missing userID in path params" });
+            return;
+        }
+
+        const {data: user, error } = await supabase
+            .from('users')
+            .select('userID, email, username, firstname, lastname, role')
+            .eq('userID', userID)
+            .single();
+        
+        if (error) {
```

Comment:
It seems we write these a lot. It's okay but for future reference I would make a function to handle the errors and no users found.